### ISSUE TITLE
Getting a Content size exceeds specified contentLength. error when after upgrading to v0.8.6

### ISSUE THREAD

**Issue by bettdouglas** [dart-lang/gcloud#144]
After upgrading to v0.8.6, I start getting this error. 0.8.5 works without any errors

```lang=dart
  Future<String> uploadFile(File file, String fileName) async {
    // https://console.cloud.google.com/apis/dashboard storage needs to be enabled

    fileName = '${Uuid().v4()}-$fileName'.replaceAll(' ', '_');
    await file.openRead().pipe(bucket.write(fileName));
    return fileName;
  }
```

```
  Content size exceeds specified contentLength. 2335 bytes written while expected 262. [QSBzZXJ2ZXIgYXBwIGJ1aWx0IHVzaW5nIFtTaGVsZl0oaHR0cHM6Ly9wdWIuZGV2L3BhY2thZ2VzL3NoZWxmKSwKY29uZmlndXJlZCB0byBlbmFibGUgcnVubmluZyB3aXRoIFtEb2NrZXJdKGh0dHBzOi8vd3d3LmRvY2tlci5jb20vKS4KVGhpcyBwcm9qZWN0IHNlcnZlcyBhcyBhbiBleGFtcGxlIG9mIGhvdyB0byB1c2UgR29vZ2xlIENsb3VkIFN0b3JhZ2UuIEl0J3MgbWVhbnQgdG8gZXhwb3NlIGEgZmlsZSB1cGxvYWQgZW5kcG9pbnQgYW5kIGEgZmlsZSBkb3dubG9hZCBlbmRwb2ludC4gCgpUaGUgCgpUaGlzIHNhbXBsZSBjb2RlIGhhbmRsZXMgSFRUUCBHRVQgcmVxdWVzdHMgdG8gYC9gIGFuZCBgL2VjaG8vPG1lc3NhZ2U+YAoKIyBSdW5uaW5nIHRoZSBzYW1wbGUKCiMjIFJ1bm5pbmcgd2l0aCB0aGUgRGFydCBTREsKCllvdSBjYW4gcnVuIHRoZSBleGFtcGxlIHdpdGggdGhlIFtEYXJ0IFNES10oaHR0cHM6Ly9kYXJ0LmRldi9nZXQtZGFydCkKbGlrZSB0aGlzOgoKYGBgCiQgZGFydCBydW4gYmluL3NlcnZlci5kYXJ0ClNlcnZlciBsaXN0ZW5pbmcgb24gcG9ydCA4MDgwCmBgYAoKQW5kIHRoZW4gZnJvbSBhIHNlY29uZCB0ZXJtaW5hbDoKYGBgCiQgY3VybCBodHRwOi8vMC4wLjAuMDo4MDgwCkhlbGxvLCBXb3JsZCEKJCBjdXJsIGh0dHA6Ly8wLjAuMC4wOjgwODAvZWNoby9JX2xvdmVfRGFydApJX2xvdmVfRGFydApgYGAKCiMjIFJ1bm5pbmcgd2l0aCBEb2NrZXIKCklmIHlvdSBoYXZlIFtEb2NrZXIgRGVza3RvcF0oaHR0cHM6Ly93d3cuZG9ja2VyLmNvbS9nZXQtc3RhcnRlZCkgaW5zdGFsbGVkLCB5b3UKY2FuIGJ1aWxkIGFuZCBydW4gd2l0aCB0aGUgYGRvY2tlcmAgY29tbWFuZDoKCmBgYAokIGRvY2tlciBidWlsZCAuIC10IG15c2VydmVyCiQgZG9ja2VyIHJ1biAtaXQgLXAgODA4MDo4MDgwIG15c2VydmVyClNlcnZlciBsaXN0ZW5pbmcgb24gcG9ydCA4MDgwCmBgYAoKQW5kIHRoZW4gZnJvbSBhIHNlY29uZCB0ZXJtaW5hbDoKYGBgCiQgY3VybCBodHRwOi8vMC4wLjAuMDo4MDgwCkhlbGxvLCBXb3JsZCEKJCBjdXJsIGh0dHA6Ly8wLjAuMC4wOjgwODAvZWNoby9JX2xvdmVfRGFydApJX2xvdmVfRGFydApgYGAKCllvdSBzaG91bGQgc2VlIHRoZSBsb2dnaW5nIHByaW50ZWQgaW4gdGhlIGZpcnN0IHRlcm1pbmFsOgpgYGAKMjAyMS0wNS0wNlQxNTo0NzowNC42MjA0MTcgIDA6MDA6MDAuMDAwMTU4IEdFVCAgICAgWzIwMF0gLwoyMDIxLTA1LTA2VDE1OjQ3OjA4LjM5MjkyOCAgMDowMDowMC4wMDEyMTYgR0VUICAgICBbMjAwXSAvZWNoby9JX2xvdmVfRGFydApgYGAKIyBDbG91ZCBSdW4KIyMjIEJ1aWxkaW5nIHRoZSBkb2NrZXIgaW1hZ2UKYGBgZ2Nsb3VkIGJ1aWxkcyBzdWJtaXQgLS10YWcgZ2NyLmlvL2Zvci10aGUtY29tbXVuaXR5L3VuaXRlY2gtZmlsZS1zZXJ2ZXJgYGAKCiMjIyBEZXBsb3lpbmcgdG8gY2xvdWQgcnVuCmBgYGdjbG91ZCBydW4gZGVwbG95IC0taW1hZ2UgZ2NyLmlvL2Zvci10aGUtY29tbXVuaXR5L3VuaXRlY2gtZmlsZS1zZXJ2ZXIgLS1taW4taW5zdGFuY2VzIDEgLS1pbmdyZXNzIGFsbCAtLWFsbG93LXVuYXV0aGVudGljYXRlZGBg]
  package:http/src/io_client.dart 90:7                         IOClient.send
  ===== asynchronous gap ===========================
  package:googleapis_auth/src/auth_http_utils.dart 36:22       AuthenticatedClient.send
  ===== asynchronous gap ===========================
  package:_discoveryapis_commons/src/api_requester.dart 69:20  ApiRequester.request
  ===== asynchronous gap ===========================
  package:googleapis/storage/v1.dart 2765:23                   ObjectsResource.insert
  ===== asynchronous gap ===========================
  package:gcloud/src/storage_impl.dart 640:15                  _MediaUploadStreamSink._startNormalUpload.<fn>
  
```

The example here https://github.com/bettdouglas/firestore_proxy has a simple reproducible example. Seems like there was a new bug introduced with the upgrade to googleapis.

**Comment by zemmyang** [comment]
got the same issue - my code is exactly the same as before, i'm sending the exact same files. 

larger files (~20MB) are sent just fine, whereas smaller files (~40KB) run into the content length error

downgrading to 0.8.5 fixed it

**Comment by Fraa-124** [comment]
Same issue, can you please take a look at this? @kevmoo

**Comment by kevmoo** [comment]
Hey @jonasfj – thoughts?

**Comment by jonasfj** [comment]
@simolus3 sounds like this is related to https://github.com/dart-lang/gcloud/pull/134, I don't see any other changes related to this that have gone into `0.8.6`.

**Comment by jonasfj** [comment]
@zemmyang does it add up to files being larger than 1MB working fine, and smaller files not working?

Because that would suggest that normal uploads / non-resumable uploads are broken:
https://github.com/dart-lang/gcloud/blob/1a01b1769da2cf9cf189760e3a757ea16426bd53/lib/src/storage_impl.dart#L538-L542

**Comment by jonasfj** [comment]
Proposed fix in https://github.com/dart-lang/gcloud/pull/147

**Comment by jonasfj** [comment]
Fix released in 0.8.7
